content/en/blog/kernel-testing.md

```diff
@@ -0,0 +1,172 @@
+* Effortless distribution of VM images: We should ensure easy sharing and deployment of the virtual machine images.
+* Parallel execution of tests on multiple VMs: Tests should run concurrently on each virtual machine to expedite the process.
+* Reproducibility in local environments for debugging purposes: Developers should be able to replicate the test environment locally to investigate and troubleshoot issues.
+* Straightforward and user-friendly presentation of test results: The test results should be presented in a simple and intuitive manner to immediately spot failures.
+
+#### Ignite a Firecracker microVM
+
+[Weave Ignite](https://https://github.com/weaveworks/ignite) is used to provision the [firecracker](https://github.com/firecracker-microvm/firecracker) microVMs. Weave Ignite is an open-source tool designed for lightweight and fast virtual machine management. It enables users to effortlessly create and manage virtual machines (VMs) for various purposes, such as development, testing, and experimentation. 
+One of the main reasons why we chose to use this tool was its capability to create firecracker microVMs from kernels and rootfs packed as OCI images.
+Currently, we are using a patched version located at [a forked repository](https://github.com/therealbobo/ignite). These patches were essential to enable the booting of kernels that necessitated the use of an initrd (initial ramdisk).
+
+#### Kernel & Rootfs OCI images
+
+Virtual machines consist of two essential layers: the kernel and the rootfs. These layers are packaged and distributed as OCI (Open Container Initiative) images. The kernel image encompasses the kernel that the virtual machine relies on, in contrast the rootfs image serves as the fundamental building block of a virtual machine, offering the essential filesystem necessary for booting the VM. Typically, these rootfs images incorporate a Linux distribution. 
+For more info on how we build them please check the available [images documentation](https://github.com/falcosecurity/kernel-testing/tree/main/images).
+
+#### Ansible Playbooks
+
+Automation is accomplished through the utilization of [Ansible](https://docs.ansible.com/ansible/latest/index.html). A collection of [playbooks](https://github.com/falcosecurity/kernel-testing/tree/main/ansible-playbooks) is responsible for:
+
+* Orchestrating the provisioning of microVMs.
+* Configuring the machines.
+* Retrieving the code to be tested.
+* Eliminating the microVMs once the testing process is completed.
+
+#### Results Presentation
+
+We wanted the test data to be publicly and easily accessible by anyone, thus we had to find a way to represent the test output.  
+Since there are 3 possible ways of instrumenting the kernel, that is using a kernel module or one of the available eBPF probes, the playbooks perform up to 3 tests. Taking into account that the modern eBPF probe is built in the Falco libraries, only 2 drivers need to be compiled.
```

Comment:
```suggestion
Since there are 3 possible ways of instrumenting the kernel, that are using a kernel module or one of the available eBPF probes, the playbooks perform up to 3 tests. Taking into account that the modern eBPF probe is built in the Falco libraries, only 2 drivers need to be compiled.
```